bin/geoserver.fn.sh

```diff
@@ -92,72 +92,39 @@ stop_geoserver() {
 
 # Installs Java and webapp_runner
 #
-# Usage: install_webapp_runner <build_dir> <cache_dir> <java_version> <webapp_runner_version>
+# Usage: install_java_webapp_runner <build_dir> <cache_dir> <env_dir>
 #
-install_webapp_runner() {
-    local jvm_url
-    local runner_url
+install_java_webapp_runner() {
+    local b_dir
+    local c_dir
+    local e_dir
 
-    local build_d
-    local cache_d
-
-    local tmp_d
-    local jre_version
-    local runner_version
-
-    local cached_jvm_common
-    local cached_runner
-
-    build_d="${1}"
-    cache_d="${2}"
-    jre_version="${3}"
-    runner_version="${4}"
-
-    jvm_url="https://buildpacks-repository.s3.eu-central-1.amazonaws.com/jvm-common.tar.xz"
-    runner_url="https://buildpacks-repository.s3.eu-central-1.amazonaws.com/webapp-runner-${runner_version}.jar"
+    local java_war_buildpack_url
+    local java_war_buildpack_dir
 
-    # Install JVM common tools:
-    cached_jvm_common="${cache_d}/jvm-common.tar.xz"
+    b_dir="${1}"
```

Comment:
```suggestion
    build_dir="${1}"
```

Please use explicit names `build_dir`, `cache_dir`, `env_dir`, it doesn't cost much and it helps our brains

Reply:
This won't work since `build_dir` and `cache_dir` are defined as `readonly` at the beginning of the `bin/compile` script.

Maybe `build_d`, `cache_d` and `env_d` would be OK? (at least it would make things a bit more consistent since we use these names in other functions)

Reply:
No I disagree, it must be consistent across buildpacks are `readonly` variables from the main global to all functions called? (bash is so terrible)

In this case I would be in favor of defining global variable from the `compile` as `BUILD_DIR` (usually capital variables are for global env variables, which they seem to be)

Reply:
All right, I understand that you disagree but I'm not sure to understand how you want it to be :-/

So here's my suggestion:

1. Make `BUILD_DIR`, `CACHE_DIR` and `ENV_DIR` `readonly` in `bin/compile` **before** declaring any functions;
2. Use them for what they are (readonly global), also in functions (so don't pass them as functions parameters anymore).

Would that be OK?